controllers/keda/scaledobject_controller.go

```diff
@@ -58,6 +58,7 @@ import (
 // +kubebuilder:rbac:groups=autoscaling,resources=horizontalpodautoscalers,verbs="*"
 // +kubebuilder:rbac:groups="",resources=configmaps;configmaps/status,verbs=get;list;watch
 // +kubebuilder:rbac:groups="",resources=events,verbs="*"
+// +kubebuilder:rbac:groups="",resources=serviceaccounts/token,verbs=create;get
```

Comment:
Should we include this as default or disable as default and explain how to enable this @kedacore/keda-core-contributors ? This setting as default, allows KEDA's service account to request tokens on behalf of any other service account in the cluster, which could mean a privilege escalation.

Users should define the `resourceNames` if they want to reduce the exposition of the RBAC setting-> https://kubernetes.io/docs/reference/access-authn-authz/rbac/#referring-to-resources
We can (and we must) handle this option in helm as we do with the * of the custom resources, but I'm not sure about the best approach here

Reply:
Maybe we can restrict it to an environment variable? I know for secrets, there exists the variable `KEDA_CLUSTER_OBJECT_NAMESPACE`, but I'm not exactly sure how it works. Is there existing code that allows us to use something like `WATCH_NAMESPACE` as a comma list of namespaces that keda is allowed to create/get service account tokens for?

Reply:
I think that we can just limit this using RBAC and KEDA's code doesn't need to know about this. We added the flag for secrets because we get secrets across all the cluster and the lister/informer generates a cluster scoped cache. In this case as the TokenRequest isn't a cached API, I think that we don't need to inform to KEDA about limited accesses, it will just fail if the RBAC blocks the request.
The RBAC protection can be done just removing the default permission and documenting how to add RBAC permission for a given service account ([using `resourceNames`](https://kubernetes.io/docs/reference/access-authn-authz/rbac/#referring-to-resources)) or globally. As the end users will be who deploy that RBAC, the decision will be on their own

Reply:
Cool, makes sense to me, thanks!